CitizenLabDotCo/citizenlab
# [CL-1957] Copy content_builder_layouts when copy Project

Copies `content_builder_layouts` associated with source project, when copying a project either locally (`LocalProjectCopyService`), or via the Project Copy Tool (`ProjectCopyService`).

Also copies any `content_builder_layout_images` (and the related image files) associated with copied layout(s), and ensures their `:code` values match those created in the copied layout(s) to refer to the new images.

Includes tests of the new functionality.

Does not include adding `content_builder_layouts` & `content_builder_layout_images` to tenant template [serializer](https://github.com/CitizenLabDotCo/citizenlab/blob/master/back/engines/commercial/multi_tenancy/app/services/multi_tenancy/templates/serializer.rb), which will be addressed separately in [CL-2746](https://citizenlab.atlassian.net/browse/CL-2746)

Manual testing:
Copies `content_builder_layouts` + (apparently) any number of `layout_images`, with images at any position in layout.
Appears to work with Project copy Tool

[CL-2746]: https://citizenlab.atlassian.net/browse/CL-2746?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
